src/scripts/prefill.ts

```diff
@@ -0,0 +1,56 @@
+import 'dotenv/config';
+import { readFile } from 'fs/promises';
+import { getTextsBeforeDones } from '../helpers';
+import { insertToSupabase } from '../supabase';
+
+interface TgJsonExport {
+  name: string;
+  id: number;
+  messages: [
+    {
+      date_unixtime: string;
+      from?: string;
+      from_id?: string;
+      text: '' | ['' | { type: string; text: string }];
+    }
+  ];
+}
+
+async function main() {
+  const file = await readFile(process.argv[2], 'utf-8');
+  const json: TgJsonExport = JSON.parse(file);
+
+  const doneRawMessages = json.messages.map((message) => {
+    if (!(message.text instanceof Array)) {
+      return null;
+    }
+    let stringifiedText = message.text
+      .map((part) => {
+        if (part instanceof Object) {
+          return part.text;
+        } else {
+          return part;
+        }
```

Comment:
```suggestion
       .map((part) => {
        return typeof part === 'string' ? part : part.text;
      })
```
Let's remove checks related to instances, etc. Basically, what you need to know, is your `part` contains `text` property.